server/src/main/java/com/microsoft/java/bs/core/internal/gradle/GradleApiConnector.java

```diff
@@ -197,7 +197,16 @@ public StatusCode runTests(URI projectUri,
             launcher.withArguments(args);
             launcher.setJvmArguments(jvmOptions);
             // env vars requires Gradle >= 3.5
-            launcher.setEnvironmentVariables(envVars);
+            if (envVars != null) {
+              // Running Gradle tests on Windows seems to require the SystemRoot env var
+              // Otherwise Windows complains "Unrecognized Windows Sockets error: 10106"
+              // Would check Windows using org.gradle.internal.os.OperatingSystem but it's internal
+              String systemRoot = System.getenv("SystemRoot");
```

Comment:
I don't know.  Maybe.  Would be nice if the BSP spec and Gradle had both `envVars` and `additionalEnvVars`.
I can change it to set all env vars if you want.

Reply:
Maybe we can have a try:

1. Get existing envs and append customized ones
2. Remove the disable flag for those two env tests
3. See if this can fix the issue :)